lib/zclient.c

```diff
@@ -811,6 +825,12 @@ int zapi_route_encode(uint8_t cmd, struct stream *s, struct zapi_route *api)
 					     16);
 				stream_putl(s, api_nh->ifindex);
 				break;
+			default:
```

Comment:
If we add a new nexthop type then we will silently not know about this case statement, we already get warned about a non-existent nexthop type in zebra so this is not necessary

Reply:
Fair point. I thought those sanity checks were removed by accident since the commit message didn't mention anything about it.

The only problem from my perspective is that, if the nexthop type is not set (zero), we won't catch the problem in the encode/decode functions since `0` is not a valid nexthop type. Not a big problem since zebra detects the problem later as you said, but ideally I think we should perform sanity checks at multiple levels and detect inconsistencies as early as possible. I'll check later if we can do that without using a `default` switch statement, otherwise drop these changes.